system/web/context/Response.cfc

```diff
@@ -149,6 +215,63 @@ component accessors="true" {
 		variables.messages.addAll( arguments.message );
 		return this;
 	}
+	
+	/**
+	 * Sets the status code with a statusText for the API response
+	 * @statusCode The status code to be set
+	 * @statusText The status text to be set
+	 *
+	 * @return Returns the Response object for chaining
+	 */
+	function setStatusCode( required statusCode, statusText ) {
+		if ( isNull( arguments.statusText ) ) {
+			if ( structKeyExists( variables.STATUS_TEXTS, arguments.statusCode ) ) {
+				arguments.statusText = variables.STATUS_TEXTS[ arguments.statusCode ];
+			} else {
+				arguments.statusText = "";
+			}
+		}
+		variables.statusCode = arguments.statusCode;
+		variables.statusText = arguments.statusText;
+		return this;
+	}
+
+	/**
+	 * Sets the data and pagination from a struct with `results` and `pagination`.
+	 *
+	 * @data           The struct containing both results and pagination.
+	 * @resultsKey     The name of the key with the results.
+	 * @paginationKey  The name of the key with the pagination.
+	 *
+	 * @return Returns the Response object for chaining
+	 */
+	function setDataWithPagination( data, resultsKey = "results", paginationKey = "pagination" ) {
+		variables.data = arguments.data[ arguments.resultsKey ];
+		variables.pagination = arguments.data[ arguments.paginationKey ];
```

Comment:
use the `setPagination()` here, so we can get consistency in the struct being sent, also to make it ACF safe.